infra/docker/test-runner/conftest.py

```diff
@@ -0,0 +1,4 @@
+def pytest_addoption(parser):
+    parser.addoption("--core_url", action="store", default="core:6565")
+    parser.addoption("--serving_url", action="store", default="serving:6566")
+    parser.addoption("--allow_dirty", action="store", default="True")
```

Comment:
Is there a reason why you are setting allow-dirty to be true?

Reply:
My intention is to run the e2e test as follows:
docker-compose -f docker-compose.yml -f docker-compose.override.yml -f docker-compose.e2e.yml run e2e

That way, the e2e test run is entirely self contained. However, the above command will keep the services running even after the e2e test run finished. So for convenience i simply set allow dirty equals to true, otherwise we will need to restart the other services everytime. But i do agree that i should have probably set it to false, for consistency and accuracy of test.

That being say, this e2e test setup will be invalidated once we added test for batch retrieval, because my docker-compose setup has not included batch serving yet, only online serving.

Reply:
The only reason I added that flag was for safety, just in case somebody accidentally runs the test while connected to a production cluster. If you think that situation won't occur, then I guess its fine to make it true.